### DISCUSSION TITLE
Don't use --reload if uploading or saving multiple files at once

### DISCUSSION THREAD

**Discussion by sixth-instinct** [fastapi/fastapi#2863] [Questions] [answered]
Just a tip...

If you're uploading files and saving multiple files to your folder within your app at once, make sure you don't use uvicorn **--reload**. If you do, uvicorn will reload your scripts as soon as after the first file is saved, interrupting the uploading or saving of the next file(s), resulting in partial, corrupted saving of the file(s) and producing a network fetch error in Swagger.

It took me a while to figure this out because there was no error in the terminal output. Perhaps the fastapi tutorial could mention this.

Alternatively, you could tell uvicorn which folders within your app to watch to trigger the reload and not include the folder for your file uploads:

e.g. >uvicorn main:app --reload --reload-dir db --reload-dir main --reload-dir router

I wish uvicorn had a parameter to tell which folder to exclude e.g. --reload-exclude-dir, so I don't have to type in every single folder to include.

**Comment by ArcLightSlavik** [comment]
Have you tried this with gunicorn + uvicorn worker and it's reload?

**Comment by baby5** [comment]
specify folder out of your app...

**Comment by ivsanro1** [comment]
I also wish this functionality of `--reload-exclude-dir` existed

**Comment by tiangolo** [comment, marked as answer]
> ...interrupting the uploading or saving of the next file...

I'm not sure I understand this. :thinking:

Not sure how are you uploading files, which files, from where to where, in which way...

Are you uploading your own code somewhere? Or are you sending files to your own API that are saved under the same directory or something like that?

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
